core/templates/dev/head/domain/skill/ConceptCardBackendApiService.ts

```diff
@@ -48,69 +48,83 @@ oppia.factory('ConceptCardBackendApiService', [
+      for (var skillId in skillIds) {
+        if (!_conceptCardCache.hasOwnProperty(skillId)) {
+          return false;
+        }
+      }
+      return true;
     };
 
     return {
       /**
-       * Retrieves the concept card of a skill from the backend given a skill
-       * ID. This returns a promise object that allows a success and rejection
-       * callbacks to be registered. If the concept card is successfully loaded
-       * and a success callback function is provided to the promise object, the
-       * success callback is called with the concept card passed in as a
-       * parameter. If something goes wrong while trying to fetch the
-       * concept card, the rejection callback is called instead, if present. The
-       * rejection callback function is passed the error that occurred and the
-       * skill ID.
+       * Retrieves the concept cards of skills from the backend given the skill
+       * IDs. This returns a promise object that allows a success and rejection
+       * callbacks to be registered. If the concept cards are successfully
+       * loaded and a success callback function is provided to the promise
+       * object, the success callback is called with the concept card passed
+       * in as a parameter. If something goes wrong while trying to fetch the
+       * concept cards, the rejection callback is called instead, if present.
+       * The rejection callback function is passed the error that occurred and
+       * the skill IDs.
        */
-      fetchConceptCard: function(skillId) {
+      fetchConceptCards: function(skillIds) {
         return $q(function(resolve, reject) {
-          _fetchConceptCard(skillId, resolve, reject);
+          _fetchConceptCards(skillIds, resolve, reject);
         });
       },
 
       /**
-       * Behaves in the exact same way as fetchConceptCard (including callback
+       * Behaves in the exact same way as fetchConceptCards (including callback
        * behavior and returning a promise object), except this function will
-       * attempt to see whether the given concept card has already been loaded.
-       * If it has not yet been loaded, it will fetch the concept card from the
-       * backend. If it successfully retrieves the concept card from the
-       * backend, it will store it in the cache to avoid requests from the
-       * backend in further function calls.
+       * attempt to see whether the given concept cards have already been
+       * loaded. If they have not yet been loaded, it will fetch the concept
+       * cards from the backend. If it successfully retrieves the concept cards
+       * from the backend, it will store them in the cache to avoid requests
+       * from the backend in further function calls.
        */
-      loadConceptCard: function(skillId) {
+      loadConceptCards: function(skillIds) {
         return $q(function(resolve, reject) {
-          if (_isCached(skillId)) {
+          if (_isCached(skillIds)) {
```

Comment:
This will fail anytime even one of the skill IDs is not cached and it will end up fetching concept cards for all the skill IDs (even if one of the skill IDs was already cached). We should be doing the `_isCached()` check per skill ID. 
If the skill ID is cached, push the cached value into conceptCards. If its not cached, add it to a list of skill IDs with uncached concept cards (lets say uncachedSkillIds). Then do `_fetchConceptCards(uncachedSkillIds, function(conceptCards) ....`

Reply:
Done.